eidas-saml-parser/src/main/java/uk/gov/ida/notification/eidassaml/saml/validation/components/AssertionConsumerServiceValidator.java

```diff
@@ -1,71 +1,33 @@
 package uk.gov.ida.notification.eidassaml.saml.validation.components;
 
 import com.google.common.base.Strings;
-import net.shibboleth.utilities.java.support.resolver.CriteriaSet;
-import net.shibboleth.utilities.java.support.resolver.ResolverException;
-import org.opensaml.core.criterion.EntityIdCriterion;
-import org.opensaml.saml.common.xml.SAMLConstants;
-import org.opensaml.saml.metadata.resolver.MetadataResolver;
 import org.opensaml.saml.saml2.core.AuthnRequest;
-import org.opensaml.saml.saml2.metadata.AssertionConsumerService;
-import org.opensaml.saml.saml2.metadata.EntityDescriptor;
-import org.opensaml.saml.saml2.metadata.SPSSODescriptor;
 import uk.gov.ida.notification.exceptions.authnrequest.InvalidAuthnRequestException;
-import uk.gov.ida.notification.shared.logging.ProxyNodeLogger;
+import uk.gov.ida.notification.shared.proxy.MetatronProxy;
 
-import java.util.Collection;
-import java.util.List;
-import java.util.logging.Level;
-import java.util.stream.Collectors;
-
-import static java.text.MessageFormat.format;
+import java.net.URI;
 
 public class AssertionConsumerServiceValidator {
 
-    private final MetadataResolver metadataResolver;
+    private MetatronProxy metatronProxy;
 
-    public AssertionConsumerServiceValidator(MetadataResolver metadataResolver) {
-        this.metadataResolver = metadataResolver;
+    public AssertionConsumerServiceValidator(MetatronProxy metatronProxy) {
+        this.metatronProxy = metatronProxy;
     }
 
     public void validate(AuthnRequest request) {
         String assertionConsumerServiceURL = request.getAssertionConsumerServiceURL();
-
         if (Strings.isNullOrEmpty(assertionConsumerServiceURL)) {
             return;
         }
```

Comment:
if a request has no assertionConsumerServiceURL we should throw a 400

Reply:
Makes sense - I used the existing logic so good spot!